lib/checksizeof.cpp

```diff
@@ -317,6 +317,46 @@ void CheckSizeof::sizeofCalculationError(const Token *tok, bool inconclusive)
                 "sizeofCalculation", "Found calculation inside sizeof().", CWE682, inconclusive);
 }
 
+//-----------------------------------------------------------------------------
+
+void CheckSizeof::sizeofFunction()
+{
+    if (!_settings->isEnabled(Settings::WARNING))
+        return;
+
+    for (const Token *tok = _tokenizer->tokens(); tok; tok = tok->next()) {
+        if (Token::simpleMatch(tok, "sizeof (")) {
+
+            // ignore if the `sizeof` result is cast to void inside a macro, i.e. the calculation is
+            // expected to be parsed but skipped, such as in a disabled custom ASSERT() macro
+            if (tok->isExpandedMacro() && tok->previous()) {
+                const Token *cast_end = (tok->previous()->str() == "(") ? tok->previous() : tok;
+                if (Token::simpleMatch(cast_end->tokAt(-3), "( void )") ||
+                    Token::simpleMatch(cast_end->previous(), "static_cast<void>")) {
+                    continue;
+                }
+            }
+
+            for (const Token *argument = tok->next()->astOperand2(); argument; argument = argument->astOperand2()) {
+                const Token *checkToken = argument->previous();
+                if (checkToken->tokType() == Token::eName)
+                    break;
+                const Function * fun = checkToken->function();
+                // Dont report error if the function is overloaded
+                if (fun && fun->nestedIn->functionMap.count(checkToken->str()) == 1) {
+                    sizeofFunctionError(tok, false);
+                }
+            }
+        }
+    }
+}
+
+void CheckSizeof::sizeofFunctionError(const Token *tok, bool inconclusive)
```

Comment:
The `inconclusive` argument can be removed, it's always false. Unless you plan to start using it someday.